pyomo/core/base/block.py

```diff
@@ -821,7 +821,7 @@ def transfer_attributes_from(self, src):
                or k in self._decl:
                 setattr(self, k, src_raw_dict[k])
 
-    def _add_temporary_set(self, val):
+    def _add_implicit_sets(self, val):
         """TODO: This method has known issues (see tickets) and needs to be
         reviewed. [JDS 9/2014]"""
```

Comment:
Is this TODO still relevant? Seems to pre-date the move to GitHub

Reply:
Yes.  This is still an issue.  I am separately working on changing how to resolve it.  I didn't want to further complicate this PR with that change.  The problem is documented in, e.g., #45.